File: app/src/main/java/com/sorcerer/sorcery/iconpack/SettingsActivity.java
package com.sorcerer.sorcery.iconpack;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.sorcerer.sorcery.iconpack.settings.SettingsFragment;
import com.sorcerer.sorcery.iconpack.ui.activities.base.BaseFragmentSubActivity;

public class SettingsActivity extends BaseFragmentSubActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setToolbarBackIndicator();
    }

    @Override
    protected int provideFragmentContainer() {
        return R.id.frameLayout_settings_fragment_container;
    }

    @Override
    protected Fragment provideInitFragment() {
        return SettingsFragment.newInstance();
    }

    @Override
    protected int provideLayoutId() {
        return R.layout.activity_settings;
    }
}

File: app/src/main/java/com/sorcerer/sorcery/iconpack/SettingsActivity.kt
package com.sorcerer.sorcery.iconpack

import android.os.Bundle
import android.support.v4.app.Fragment
import com.sorcerer.sorcery.iconpack.settings.SettingsFragment
import com.sorcerer.sorcery.iconpack.ui.activities.base.BaseFragmentSubActivity

class SettingsActivity : BaseFragmentSubActivity() {

    override fun provideLayoutId(): Int {
        return R.layout.activity_settings
    }

    override fun provideFragmentContainer(): Int {
        return R.id.frameLayout_settings_fragment_container
    }

    override fun provideInitFragment(): Fragment {
        return SettingsFragment.newInstance()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setToolbarBackIndicator()
    }

}
